### PR TITLE
Share skyloong/dt40 keyboard

### PR DESCRIPTION
**Pull request by JZ-Skyloong**
## Description

Share a new rgb matrix keyboard -- skyloong/dt40. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

Resubmit

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/skyloong/dt40/info.json [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    },
+
+
+
+    "ws2812": {
+        "pin": "B4"
+    },
+    "rgb_matrix":{
+        "driver":"WS2812",
+        "led_count": 44,
```
Comment (zvecr) [inline keyboards/skyloong/dt40/info.json:93]: Isnt a thing
```suggestion
```

File: keyboards/skyloong/dt40/rules.mk [hunk 2]
```diff
@@ -0,0 +1,2 @@
+WS2812_DRIVER = pwm
+
```
Comment (zvecr) [inline keyboards/skyloong/dt40/rules.mk:2]:
```suggestion
# This file intentionally left blank
```

File: keyboards/skyloong/dt40/dt40.c [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+void board_init(void) {
+    AFIO->MAPR |= AFIO_MAPR_USART1_REMAP | AFIO_MAPR_TIM3_REMAP_PARTIALREMAP;
+
+}
+
+
+
+
+
```
Comment (zvecr) [inline keyboards/skyloong/dt40/dt40.c:59]:
```suggestion
```

File: keyboards/skyloong/dt40/halconf.h [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#pragma once
+
+#define HAL_USE_PWM TRUE
+
```
Comment (zvecr) [inline keyboards/skyloong/dt40/halconf.h:21]:
```suggestion
```

File: keyboards/skyloong/dt40/dt40.c [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+
```
Comment (zvecr) [inline keyboards/skyloong/dt40/dt40.c:18]:
```suggestion
```

File: keyboards/skyloong/dt40/halconf.h [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
```
Comment (zvecr) [inline keyboards/skyloong/dt40/halconf.h:17]:
```suggestion
```

File: keyboards/skyloong/dt40/config.h [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#pragma once
+
```
Comment (zvecr) [inline keyboards/skyloong/dt40/config.h:19]:
```suggestion
#pragma once
```

File: keyboards/skyloong/dt40/config.h [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS	//Enables RGB_MATRIX_JELLYBEAN_RAINDROPS
+#define ENABLE_RGB_MATRIX_HUE_BREATHING	//Enables RGB_MATRIX_HUE_BREATHING
+#define ENABLE_RGB_MATRIX_HUE_PENDULUM	//Enables RGB_MATRIX_HUE_PENDULUM
+#define ENABLE_RGB_MATRIX_HUE_WAVE	//Enables RGB_MATRIX_HUE_WAVE
+#define ENABLE_RGB_MATRIX_PIXEL_FRACTAL	//Enables RGB_MATRIX_PIXEL_FRACTAL
+#define ENABLE_RGB_MATRIX_PIXEL_RAIN	//Enables RGB_MATRIX_PIXEL_RAIN
+
+//Set default RGB matrix effect
+#define RGB_MATRIX_DEFAULT_MODE  RGB_MATRIX_CYCLE_LEFT_RIGHT  //Set the default mode, if none has been set
+//#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 200  //Limits maximum brightness of LEDs to 200 out of 255. If not defined maximum brightness is set to 255
```
Comment (zvecr) [inline keyboards/skyloong/dt40/config.h:62]:
```suggestion
```

File: keyboards/skyloong/dt40/config.h [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+#define RGB_MATRIX_DEFAULT_VAL 150  //Sets the default mode, if none has been set
+#define RGB_TRIGGER_ON_KEYDOWN  //Triggers RGB keypress events on key down
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS // enable framebuffer effects
+#define RGB_MATRIX_TYPING_HEATMAP_DECREASE_DELAY_MS 1
+
+//Set WS2812 driver
+#define WS2812_TRST_US 200  //The Trst period between frames
+#define WS2812_BYTE_ORDER WS2812_BYTE_ORDER_GRB  // The LEDs Byte Order
```
Comment (zvecr) [inline keyboards/skyloong/dt40/config.h:70]:
```suggestion
```

File: keyboards/skyloong/dt40/info.json [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+                { "matrix": [3, 10], "x": 13, "y": 3 }
+
+            ]
+        }
+    },
+
+
+
+    "ws2812": {
+        "pin": "B4"
```
Comment (zvecr) [inline keyboards/skyloong/dt40/info.json:89]:
```suggestion
        "driver": "pwm",
        "pin": "B4"
```

**Review by zvecr — CHANGES_REQUESTED** [review]
In its current form, this PR cannot be accepted due to the submodule changes in `lib/chibios`.

**Comment by zvecr** [comment]
Also, `info.json` layout would be considered broken as it does not correctly represent the product,
```
linting skyloong/dt40
Ψ Lint check passed!
Keyboard Name: DT40
Manufacturer: JZ-Skyloong
Website: https://github.com/JZ-Skyloong
Maintainer: JZ-Skyloong
Layouts: LAYOUT
Processor: STM32F103
Bootloader: stm32duino
Layout Layout:
┌──┐    ┌──┐┌──┐┌──┐┌──┐┌──┐┌──┐┌──┐┌──┐┌──┐┌──┐    ┌──┐
│  │    │  ││  ││  ││  ││  ││  ││  ││  ││  ││  │    │  │
└──┘    └──┘└──┘└──┘└──┘└──┘└──┘└──┘└──┘└──┘└──┘    └──┘
┌──┐    ┌──┐┌──┐┌──┐┌──┐┌──┐┌──┐┌──┐┌──┐┌──┐        ┌──┐
│  │    │  ││  ││  ││  ││  ││  ││  ││  ││  │        │  │
└──┘    └──┘└──┘└──┘└──┘└──┘└──┘└──┘└──┘└──┘        └──┘
    ┌──┐    ┌──┐┌──┐┌──┐┌──┐┌──┐┌──┐┌──┐┌──┐    ┌──┐┌──┐
    │  │    │  ││  ││  ││  ││  ││  ││  ││  │    │  ││  │
    └──┘    └──┘└──┘└──┘└──┘└──┘└──┘└──┘└──┘    └──┘└──┘
┌──┐┌──┐┌──┐┌──┐        ┌──┐        ┌──┐┌──┐┌──┐┌──┐┌──┐
│  ││  ││  ││  │        │  │        │  ││  ││  ││  ││  │
└──┘└──┘└──┘└──┘        └──┘        └──┘└──┘└──┘└──┘└──┘
```

**Comment by JZ-Skyloong** [comment]
Thank you very much for your guidance @zvecr

**Pull request closed without merging**